src/main/java/com/conveyal/r5/analyst/cluster/PathWriter.java

```diff
@@ -142,6 +142,8 @@ public void finishAndStorePaths () {
             dataOutput.writeInt(pathForIndex.size());
             for (Path path : pathForIndex) {
                 dataOutput.writeInt(path.patterns.length);
+                dataOutput.write(path.accessMode.toString().substring(0,3).getBytes());
+                dataOutput.write(path.egressMode.toString().substring(0,3).getBytes());
```

Comment:
The call as-is will be sensitive to the default encoding of the platform, and could write multiple bytes. Java characters are usually two byes and some encodings may preserve that representation. Should we switch to `getBytes(Charset charset)` using UTF-8, and further check that the output is of the expected length?

I guess there are two concerns driving truncation here - overall size of output, and the need for fixed-width fields. If the output is gzipped then frequently occurring tokens of any length are likely to be compressed to the same size, so truncation doesn't have much effect. For obtaining fixed widths, truncation to 3 characters seems somewhat arbitrary. Should we go with maximum truncation (to one byte)?

Reply:
The need to write these out as short fixed-width codes may be a good argument for assigning codes to the enum values (as was done in a previous commit). However instead of integer values, maybe we could use single uppercase letters, written out as single UTF-8 bytes. We could also just have a static `getSingleByteCode(accessMode)` method containing a case statement, which would probably be significantly faster than dynamically manipulating strings.